src/shared/version_op.c

```diff
@@ -667,7 +667,9 @@ os_info *get_unix_version()
                 len += strlen(info->os_version);
                 len += strlen(info->os_codename);
                 os_realloc(info->os_version, len, info->os_version);
-                snprintf(info->os_version, len, "%s (%s)", info->os_version, info->os_codename);
+                char tmp_os_version[len];
```

Comment:
I see two issues here:
1. Variable length array on the stack is not legal in C90, so this instruction may fail on old systems.
Please use either dynamic memory or a large enough static array. The latter is the most efficient option.
2. `strncpy()` does not guarantee writing the terminating byte (`'\0'`) if it does not fit. This would lead to an unterminated string. Please use `snprintf()` or write an explicit zero. Again, the latter is more efficient.

```c
char tmp_os_version[1024];
strncpy(tmp_os_version, info->os_version, sizeof(tmp_os_version));
tmp_os_version[sizeof(tmp_version) - 1] = '\0`;
```